secure_message/resources/messages.py

```diff
@@ -137,24 +137,42 @@ def _try_send_alert_email(message):
 
     @staticmethod
     def _inform_case_service(message):
+
         if current_app.config['NOTIFY_CASE_SERVICE'] == '1':
-            if message.msg_from == constants.BRES_USER:
-                case_user = constants.BRES_USER
+            case_user = MessageSend._resolve_user_details_for_case_service(g.user, message)
+
+            if case_user:
+                case_service.store_case_event(message.collection_case, case_user)
             else:
-                party_data = party.get_user_details(message.msg_from)  # NOQA TODO avoid 2 lookups(see validate)
-                if party_data:
-                    first_name = party_data['firstName'] if 'firstName' in party_data else ''
-                    last_name = party_data['lastName'] if 'lastName' in party_data else ''
-                    case_user = '{} {}'.format(first_name, last_name).strip()
-                    if not case_user:
-                        case_user = 'Unknown user'
-                        logger.info('no user names in party data for id  Unknown user used in case ',
-                                    party_id=party_data['id'])
-                # else not testable , as it fails on validation
-            case_service.store_case_event(message.collection_case, case_user)
+                logger.info(f'unable to resolve details, for case service, for {g.user.user_uuid} role: {g.user.role}')
+
         else:
             logger.info('Case service notifications switched off, hence not sent', msg_id=message.msg_id)
 
+    @staticmethod
+    def _resolve_user_details_for_case_service(user, message):
+        case_user = ''
+        if message.msg_from == constants.BRES_USER:
+            case_user = constants.BRES_USER
+        else:
+            if g.user.is_internal:
+                user_data = internal_user_service.get_user_details(user.user_uuid)
+                service = 'user'
+            else:
+                user_data = party.get_user_details(message.msg_from)  # NOQA TODO avoid 2 lookups(see validate)
+                service = 'party'
+
+            if user_data:
```

Comment:
No sure you need this if here. user_data is going to have always something in there

Reply:
user data can be none if it is not known in party